### PR TITLE
CE-33707: Fixed deprecated methods for PHPUnit v9.3.0

### PR DESCRIPTION
**Pull request by karyna-t**
### Description (*)
This PR fixes CE project codebase compatibility problems with the PHPUnit v.9.3.0.
See changes here: [PHPUnit 9.3.0 ChangeLog](https://github.com/sebastianbergmann/phpunit/blob/9.3.0/ChangeLog-9.3.md#configuration-of-code-coverage-and-logging-in-phpunitxml)

In this PR was changed 90 files where the main problem was using at() matcher that has been deprecated.

### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/576
https://github.com/magento/partners-magento2b2b/pull/579

### Related Issues
1. magento/magento2#33707

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests EE, Functional Tests B2B, Functional Tests CE, Integration Tests, WebAPI Tests, Static Tests

File: app/code/Magento/Quote/Test/Unit/Model/QuoteManagementTest.php [hunk 1]
```diff
@@ -416,9 +429,12 @@ public function testCreateEmptyCartForCustomerReturnExistsQuote()
         $this->model->createEmptyCartForCustomer($userId);
     }
 
-    public function testAssignCustomerFromAnotherStore()
+    /**
+     * @return void
+     */
+    public function testAssignCustomerFromAnotherStore(): void
     {
-        $this->expectException('Magento\Framework\Exception\StateException');
```
Comment (ihor-sviziev) [inline app/code/Magento/Quote/Test/Unit/Model/QuoteManagementTest.php:421]: AFAIK It was intentionally done in order not use ::class due to performance reasons - on big data sets it's noticeable when php unit tries additionally lookup for exception class.
@lbajsarowicz, am I correct?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
In general, your changes look good to me. Could you check failing tests?

**Review by andrewbess — APPROVED** [review]
This is an intermediate approval.
All requested changes will be fixed in parent PR.

**Pull request merged**